### DISCUSSION TITLE
Improve Metrics insights Text formatting for issues, discussions, etc.

### DISCUSSION THREAD

**Discussion by Andre601** [lowlighter/metrics#922] [Ideas]
Recently tried out the insights option of the metrics page.

There, I noticed that for issues, Pull requests, etc. the formatting is not always the same as in the source itself.

As an example, insights displays the following for an issue comment I made a while back:
![image](https://user-images.githubusercontent.com/11576465/155901378-cba4537a-1a37-400e-9a66-8e9f07748a6f.png)

Meanwhile, on GitHub is the issue actually looking like this:
![image](https://user-images.githubusercontent.com/11576465/155901400-b6c814f6-0760-4dbf-9a36-1496d33f0f3b.png)

A few things are noticeable:

1. Quotes aren't indicated as such. At most do they have an indent.
2. Code blocks are missing.
3. Newlines don't seem to be respected, as seen with the image.

I assume this could be because of markdown rendering? Iirc does GitHub's public events API return content as markdown in the case of comments and such...
Whatever the issue is, I hope this can be further improved as the insights option is rather nice.

**Comment by Andre601** [comment]
Something else I noticed:

If a comment contains checkboxes, either checked or unchecked, and you hover over them does your cursor show the "loading" icon despite the site no longer loading.
Is stuff still loading in the background for this to show?

**Comment by lowlighter** [comment]
> I assume this could be because of markdown rendering? Iirc does GitHub's public events API return content as markdown in the case of comments and such...

GitHub does indeed return markdown content, which means that the app needs to handle the conversion to html by itself.
We use [markedjs/marked](https://github.com/markedjs/marked) for parsing, and the css is a loose adaptation of [primer/css](https://github.com/primer/css), which may be missing some styles and maybe newer design updates, since it's been quite some time since the web app redesign.

> Code blocks are missing.

Code blocks should be doable too, though we may need to add some lazy loading rendering, because loading all highlighting syntaxes for all languages would have some scalability and performance issues.

>  Is stuff still loading in the background for this to show?

Actually insights option is just a html rendering of a json output from the regular metrics, but with a static configuration. So there's no stuff loading in the background, but it could be some remnants of the css of other page of the web app (since it shares some part of css).

The code snippet below contains the current insights configuration:
https://github.com/lowlighter/metrics/blob/3553137563208ff64cb470b8f23fd8c503627f32/source/app/metrics/index.mjs#L212-L246

>  Whatever the issue is, I hope this can be further improved as the insights option is rather nice.

Overall, these issues does not requires too much changes or refactoring I think. I'll add an issue for tracking these changes.
I may work on it the next time I'll do a small metrics backlog rush, unless someone else wants to work on it in the meantime 👍 

Glad you enjoy this feature though 😄 !
